packages/origin-247-transfer/src/issue.service.ts

```diff
@@ -0,0 +1,67 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { CertificateService, CERTIFICATE_SERVICE_TOKEN } from '@energyweb/origin-247-certificate';
+import {
+    EnergyTransferRequestRepository,
+    ENERGY_TRANSFER_REQUEST_REPOSITORY
+} from './repositories/EnergyTransferRequest.repository';
+import { EventBus } from '@nestjs/cqrs';
+import { EnergyTransferRequest, State } from './EnergyTransferRequest';
+import { chunk } from 'lodash';
+import { AwaitingValidationEvent } from './batch/validate.batch';
+import { BatchConfiguration, BATCH_CONFIGURATION_TOKEN } from './batch/configuration';
+
+@Injectable()
+export class IssueService {
+    constructor(
+        @Inject(CERTIFICATE_SERVICE_TOKEN)
+        private certificateService: CertificateService<unknown>,
+        @Inject(ENERGY_TRANSFER_REQUEST_REPOSITORY)
+        private etrRepository: EnergyTransferRequestRepository,
+        @Inject(BATCH_CONFIGURATION_TOKEN)
+        private batchConfiguration: BatchConfiguration,
+        private eventBus: EventBus
+    ) {}
+
+    public async issueTask() {
+        const etrs = await this.etrRepository.findByState(State.IssuanceAwaiting);
+
+        const etrGroups = chunk(etrs, this.batchConfiguration.issueBatchSize);
```

Comment:
Why would we want to have different batch sizes for issue and validate?

Reply:
Because those two have different charateristics. We can for example process more validations (which are entirely off-chain), but we cannot process more issues (which are off-chain). That's why default values for these configurations are different.